qds_sdk/cloud/aws_cloud.py

```diff
@@ -163,4 +177,179 @@ def create_parser(self, argparser):
                                                " to provide access to external hosts", )
         network_config_group.add_argument("--master-elastic-ip",
                                           dest="master_elastic_ip",
-                                          help="master elastic ip for cluster")
\ No newline at end of file
+                                          help="master elastic ip for cluster")
+
+
+    def get_results(self, cmd, fp=sys.stdout, storage_credentials=None, r={}, delim=None, qlog=None, include_header=None):
+        boto_conn = boto.connect_s3(aws_access_key_id=storage_credentials['storage_config']['access_key'],
+                                    aws_secret_access_key=storage_credentials['storage_config']['secret_key'])
+
+        log.info("Starting download from result locations: [%s]" % ",".join(r['result_location']))
+        # fetch latest value of num_result_dir
+        num_result_dir = Command.find(cmd.id).num_result_dir
+
+        for s3_path in r['result_location']:
+
+            # If column/header names are not able to fetch then use include header as true
+            if include_header.lower() == "true" and qlog is not None:
+                self.write_headers(qlog, fp)
+
+            # In Python 3,
+            # If the delim is None, fp should be in binary mode because
+            # boto expects it to be.
+            # If the delim is not None, then both text and binary modes
+            # work.
+
+            _download_to_local(boto_conn, s3_path, fp, num_result_dir, delim=delim,
+                               skip_data_avail_check=isinstance(self, PrestoCommand))
+
+    def write_headers(self, qlog, fp):
+        col_names = []
+        qlog = json.loads(qlog)
+        if qlog["QBOL-QUERY-SCHEMA"] is not None:
+            qlog_hash = qlog["QBOL-QUERY-SCHEMA"]["-1"] if qlog["QBOL-QUERY-SCHEMA"]["-1"] is not None else \
+            qlog["QBOL-QUERY-SCHEMA"][qlog["QBOL-QUERY-SCHEMA"].keys[0]]
+
+            for qlog_item in qlog_hash:
+                col_names.append(qlog_item["ColumnName"])
+
+            col_names = "\t".join(col_names)
+            col_names += "\n"
+
+        fp.write(col_names)
+
+
+def _read_iteratively(key_instance, fp, delim):
+    key_instance.open_read()
+    while True:
+        try:
+            # Default buffer size is 8192 bytes
+            data = next(key_instance)
+            if sys.version_info < (3, 0, 0):
```

Comment:
From line 228 to 238  code is repeated (here, download_local_to_azure/oracle). Can we form a separate function for this?
A function for delimiter code write and another for utf8 encoding (in get_contents_to_file_azure/oracle and commands.py:220).